File: neetcode-all/arrays_and_hashing/group_anagrams.cc
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> defaultdict;

        for(auto &s: strs){
            string key = s;
            sort(key.begin(), key.end());
            defaultdict[key].push_back(s);
        }
        vector<vector<string>> res;

        for(auto &d:defaultdict){
            res.push_back(d.second);
        }
        return res;
    }
};

File: neetcode-all/arrays_and_hashing/group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        #time = o(n * klogk) due to sorting 
        #space = o(n * k) to store strings in dict 
        res = defaultdict(list)

        for s in strs:
            key = tuple(sorted(s))
            res[key].append(s)
        return list(res.values())
